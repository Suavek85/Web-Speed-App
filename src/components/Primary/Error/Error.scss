@import '../../../scss/styles';

$block: error;

.#{$block} {
  position: relative;
  width: 200px;
  height: 44px;
  margin-left: 20px;
  font-size: 12px;
  line-height: 41px;
  color: $color-white;
  text-align: center;
  background-color: $color-warning-loading;

  &::first-letter {
    font-size: 17px;
  }

  &::after {
    position: absolute;
    bottom: 15.5px;
    left: -5.5px;
    width: 10px;
    height: 10px;
    content: '';
    background: inherit;
    background-color: $color-warning-loading;
    transform: rotate(45deg);
  }
}

:export {
  block: $block;
}
