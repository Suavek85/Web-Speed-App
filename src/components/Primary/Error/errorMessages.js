  export const errorMessages = [
    { status: 429, message:  `⚠️ We're busy...` },
    { status: 400, message:  `⚠️ Invalid url` },
    { status: 500, message:  `⚠️ Invalid url` },
  ]

  export const genericErrorMessage = { message: `⚠️ Oops, an error...` }