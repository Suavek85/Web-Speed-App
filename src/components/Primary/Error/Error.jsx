import React from 'react'
import { useSelector } from "react-redux"
import { errorMessages, genericErrorMessage } from './errorMessages.js'
import styles from './Error.scss'  

export default function Error() {

  const errorStatus = useSelector(state => state.stateReducer.getErrorStatus) 
  const newErrorMessage = errorMessages.filter(el => el.status === errorStatus )
  newErrorMessage.length === 0 && newErrorMessage.push(genericErrorMessage)
  const { message } = newErrorMessage[0]
  
  return (
    <>
      {errorStatus && (
        <div className={styles.block} >
          {message}
        </div>
      )} 
    </>
  )
}