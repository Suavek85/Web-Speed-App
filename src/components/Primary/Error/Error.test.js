import * as React from 'react'
import * as redux from 'react-redux'
import { shallow } from 'enzyme'
import Error from './Error'

describe('Error snapshot', () => {

  it('should match the snapshot', () => {
    const errorMessageStatus = 500
    const spy = jest.spyOn(redux, 'useSelector')
    spy.mockReturnValue(errorMessageStatus)
    const wrapper = shallow(<Error />)
    expect(wrapper).toMatchSnapshot()
  })

})

describe('Error rendering', () => {

  it('should not render error if error status is null', () => {
    const errorMessageStatus = null
    const spy = jest.spyOn(redux, 'useSelector')
    spy.mockReturnValue(errorMessageStatus)
    const wrapper = shallow(<Error />)
    expect(wrapper.find('div').exists()).toBeFalsy()
  })

  it('should render error if error status is defined', () => {
    const errorMessageStatus = 500
    const spy = jest.spyOn(redux, 'useSelector')
    spy.mockReturnValue(errorMessageStatus)
    const wrapper = shallow(<Error />)
    expect(wrapper.find('div').exists()).toBeTruthy()
  })

});