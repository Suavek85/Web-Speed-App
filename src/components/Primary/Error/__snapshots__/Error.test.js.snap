// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error snapshot should match the snapshot 1`] = `
<Fragment>
  <div>
    ⚠️ Invalid url
  </div>
</Fragment>
`;
